python/EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK/EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK.py

```diff
@@ -0,0 +1,421 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK
+Description:
+  Check whether the Amazon EBS snapshots are not publicly restorable. The rule is NON_COMPLIANT if the RestorableByUserIds field is set to 'all'.
+Trigger:
+  Periodic
+Reports on:
+  AWS::::Account
+Rule Parameters:
+  None
+Scenarios:
+  Scenario: 1
+    Given: No snapshots with RestorableByUserIds parameter set to 'all'
+     Then: Return NOT_APPLICABLE
+  Scenario: 2
+    Given: One or more snapshots with RestorableByUserIds parameter set to 'all'
+     Then: Return NON_COMPLIANT with Annotation containing SnapshotIDs
+'''
+import json
+import datetime
+import boto3
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+# Function to generate evaluations for all non compliant resources
+def generate_eval_list(snapshots, event):
+    evaluations = []
+    # Looping through all available Amazon EBS snapshots
+    for snapshot in snapshots:
+        evaluation = build_evaluation(snapshot['SnapshotId'], "NON_COMPLIANT", event, resource_type=DEFAULT_RESOURCE_TYPE, annotation="Amazon EBS Snapshot: {} is public".format(snapshot['SnapshotId']))

```

Comment:
You are using the Snapshot ID as resource ID. I am worried of unexpected results here. Better to report on Account ID, and build a comments with all the Snapshot ID. It will be cheaper for customers too.

No need for `resource_type` as it is already default